docs/userguide/joins.rst

```diff
@@ -0,0 +1,260 @@
+Join Apps
+=========
+
+Join apps allows an app to define a sub-workflow: the app can launch other apps
+and incorporate them into the main task graph. They can be specified using the
+`join_app` decorator.
+
+Usage
+-----
+
+A `join_app` looks quite like a python app, but should return a ``Future``,
+rather than a value. After the python code has run, the app invocation will not
+complete until that future has completed, and the return value of the `join_app`
+will be the return value (or exception) from the returned future.
```

Comment:
say how this is different than a parsl-decorated python_app?

Reply:
this should be explained now later on in the examples